course/models/models.py

```diff
@@ -114,6 +117,9 @@ def has_view_permission(self, user):
     def has_edit_permission(self, user):
         return user.is_teacher
 
+    def get_max_submissions(self, event):
```

Comment:
This confuses people because it seems like it's a getter got max_submission_allowed. You can even not make it a function as it's just one line. Another thing event. Type can be None that causes an error for all the questions without an event. use `if event is not None and event.type == "EXAM"`.